source/extensions/filters/network/common/redis/redis_command_stats.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/stats/scope.h"
+#include "envoy/stats/timespan.h"
+
+#include "common/stats/symbol_table_impl.h"
+
+#include "extensions/filters/network/common/redis/codec.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Common {
+namespace Redis {
+
+class RedisCommandStats {
+public:
+  RedisCommandStats(Stats::Scope& scope, const std::string& prefix, bool enabled);
+
+  Stats::Counter& counter(const Stats::StatNameVec& stat_names);
+  Stats::Histogram& histogram(const Stats::StatNameVec& stat_names);
+  Stats::CompletableTimespanPtr createCommandTimer(std::string command,
+                                                   Envoy::TimeSource& time_source);
+  Stats::CompletableTimespanPtr createAggregateTimer(Envoy::TimeSource& time_source);
+  std::string getCommandFromRequest(const RespValue& request);
+  void updateStatsTotal(const std::string& command);
+  void updateStats(const bool success, const std::string& command);
+  bool enabled() { return enabled_; }
+
+private:
+  Stats::Scope& scope_;
+  Stats::StatNameSet stat_name_set_;
+  const Stats::StatName prefix_;
+  bool enabled_;
+  const Stats::StatName upstream_rq_time_;
+  const Stats::StatName latency_;
+  const Stats::StatName total_;
+  const Stats::StatName success_;
+  const Stats::StatName error_;
+  const std::string null_metric_ = "null";
```

Comment:
`const Stats::StatName null_metric_;` and initialize in your ctor like the names above. Also for "unknown" below, which you could potentially use for unexpected commands.

Reply:
Ok, I'll make `getCommandForRequest()` return a `Stats::StatName`. Unknown makes sense will try that.